### PR TITLE
refactor(crap-core): #260 — decompose matching.rs below cognitive-8

### PR DESCRIPTION
**Pull request by cmbays**
## Summary

CRAP-8 campaign slice: decomposes the two over-8 functions in `crates/crap-core/src/domain/matching.rs` (line-range coverage matching) below cognitive-8. Pure-complexity decomposition — these are ~100% covered, so cc is the lever.

## What changed

| Function | cognitive before → after | Decomposition |
|---|---|---|
| `compute_function_coverage` | 9 → **2** | thin ratio-builder over extracted counters |
| `compute_branch_coverage` | 10 → **2** | thin ratio-builder over extracted counters |

Three in-file private helpers (each cc ≤ 2):
- `line_within_span(span, line)` — the inclusive point-in-span test (`line >= start && line <= end`), shared by both counters.
- `count_line_coverage(span, file_lines)` — covered/total over `LineCoverage` in span (covered = hits > 0).
- `count_branch_coverage(span, branches)` — covered/total over `BranchCoverage` in span (`filter_map(taken)` drops `None` exactly as the original `if let Some(taken)`).

The `total == 0` fork is deliberately kept inline per parent (function → 100.0 trivially-covered; branch → `None`), since the empty-case semantics differ. Coverage math is **byte-identical** — the inclusive boundary check moved verbatim into the predicate.

## Domain purity preserved

`matching.rs` is in the purity-enforced `domain/` layer. No non-domain import added — production imports remain only `super::types::{...}` + `std::collections::HashMap` (verified). Helpers operate exclusively on existing domain types.

## Verification

- `cargo nextest run --workspace --all-targets` — 1322/1322 pass.
- **The line-range boundary-invariant proptests all pass** (the correctness net for this file): `boundary_precision`, `boundary_inclusive_start`/`_end`, `coverage_always_0_to_100`, `covered_lte_total`, `no_cross_file_leakage`, the `branch_*` variants, and both fork guards (`no_instrumentable_lines_100_pct`, `no_branch_points_gives_none`) — 40/40 matching tests green.
- Analyzer cc re-measurement: both targets cc2, all 3 helpers ≤ 2; highest production fn in the file is `match_functions` at 5 (unchanged); nothing above 8.
- `cargo fmt --check`, `cargo +1.96.0 clippy --workspace --all-targets -- -D warnings`, `cargo +1.93 check --workspace --all-targets` (MSRV) — all green.

Part of #260 (gate flip lands with the final slice).

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: crates/crap-core/src/domain/matching.rs [hunk 1]
```diff
@@ -59,23 +59,46 @@ pub fn match_functions(
...
+/// within `span`. A line is covered when its hit count is nonzero.
+fn count_line_coverage(span: SourceSpan, file_lines: &[LineCoverage]) -> (usize, usize) {
+    let in_span = file_lines
+        .iter()
+        .filter(|line| line_within_span(span, line.line));
+
+    let total = in_span.clone().count();
+    let covered = in_span.filter(|line| line.hits > 0).count();
+    (covered, total)
+}
```
[the remark this answers is not in the document]
Comment (cmbays) [reply crates/crap-core/src/domain/matching.rs:79]: Declining — the `.fold()` bundles two counters into a tuple accumulator, which is **less** readable than the explicit `filter().count()` pair (this PR's purpose is readability decomposition). It's also a logic change in the property-tested boundary-invariant file, for a cold-path gain (per-span counting, not a hot loop). Cold-path gain + readability cost + invariant risk → decline. Not tracked.

File: crates/crap-core/src/domain/matching.rs [hunk 2]
```diff
@@ -59,23 +59,46 @@ pub fn match_functions(
...
+fn count_branch_coverage(span: SourceSpan, branches: &[BranchCoverage]) -> (usize, usize) {
+    let taken_in_span = branches
+        .iter()
+        .filter(|branch| line_within_span(span, branch.line))
+        .filter_map(|branch| branch.taken);
+
+    let total = taken_in_span.clone().count();
+    let covered = taken_in_span.filter(|&taken| taken > 0).count();
+    (covered, total)
+}
```
[the remark this answers is not in the document]
Comment (cmbays) [reply crates/crap-core/src/domain/matching.rs:93]: Declining with its sibling — same reasoning (cold-path `.fold()`, readability cost, boundary-invariant file).

**Comment by cmbays** [comment]
**Bot-review disposition.** Gemini posted 2 MEDIUM findings (`.fold()` single-pass vs the explicit `filter().count()` pair). **Declined** — the `.fold()` is less readable (counter to this PR's readability purpose), a cold-path gain, and a logic change in the property-tested boundary-invariant file. CodeRabbit creditless (did not run) — protocol **Gemini-complete**. CI green, mergeable; the 40 matching boundary-invariant tests pass.
